src/state/controls/Control.ts

```diff
@@ -104,9 +104,9 @@ export abstract class Control<T extends IControlData> extends EventEmitter imple
     /**
      * Update this control on the server.
      */
-    public update(controlUpdate: IControlUpdate): Promise<void> {
+    public update<T extends IControlUpdate>(controlUpdate: T): Promise<void> {
```

Comment:
What is the point of this, T will always implement the interface so the statements are equivalent.